libpromises/eval_context.c

```diff
@@ -1240,7 +1240,18 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         char context_copy[CF_MAXVARSIZE];
         char canonified_context[CF_MAXVARSIZE];
 
-        strcpy(canonified_context, name);
+
+        /* Redmine #7013
+         * Fix for classes names longer than CF_MAXVARSIZE. */
+        if (strlen(name) >= CF_MAXVARSIZE - 1)
+        {
+            Log(LOG_LEVEL_WARNING, "Skipping adding class [%s] as it its name "
+                "is longer than %d", name, CF_MAXVARSIZE - 1);
+            return false;
+        }
+        
+        strlcpy(canonified_context, name, CF_MAXVARSIZE);
+        
```

Comment:
``s/CF_MAXVARSIZE/sizeof(canonified_context)/``. It's more futureproof, in the sense that people changing the code later won't accidentally introduce the bug again.